internal/provider/models/sources/test/open_telemetry_metrics_test.go

```diff
@@ -0,0 +1,195 @@
+	gateway_route_id = {{.GatewayResourceID}}
+}
+{{end}}
+`
+
+func TestAccOpenTelemetryMetricsSource(t *testing.T) {
+	pipelineName := "test_parent"
+	pipelineTitle := "parent pipeline"
+	sourceName := "my_source"
+	sourceTitle := "my open telemetry metrics title"
+	sourceDesc := "my open telemetry metrics description"
+
+	resourceName := fmt.Sprintf("mezmo_open_telemetry_metrics_source.%s", sourceName)
+	resourceNamePipeline := fmt.Sprintf("mezmo_pipeline.%s", pipelineName)
+	resourceNameGateway := fmt.Sprintf("mezmo_open_telemetry_metrics_source.%s-gateway", sourceName)
+
+	config, err := ParsedAccConfig(resourceOpenTelemetryMetricsSourceConfig{
+		PipelineName:  pipelineName,
+		PipelineTitle: pipelineTitle,
+		SourceName:    sourceName,
+		SourceTitle:   sourceTitle,
+		SourceDesc:    sourceDesc,
+	}, resourceOpenTelemetryMetricsSourceConfigTpl)
+	if err != nil {
+		t.Fatalf("error parsing config template: %s", err)
+	}
+
+	configUpdate, err := ParsedAccConfig(resourceOpenTelemetryMetricsSourceConfig{
+		PipelineName:    pipelineName,
+		PipelineTitle:   pipelineTitle,
+		SourceName:      sourceName,
+		SourceTitle:     sourceTitle + " updated",
+		SourceDesc:      sourceDesc,
+		CaptureMetadata: true,
+	}, resourceOpenTelemetryMetricsSourceConfigTpl)
+	if err != nil {
+		t.Fatalf("error parsing config template: %s", err)
+	}
+
+	configGateway, err := ParsedAccConfig(resourceOpenTelemetryMetricsSourceConfig{
+		PipelineName:      pipelineName,
+		PipelineTitle:     pipelineTitle,
+		SourceName:        sourceName,
+		SourceTitle:       sourceTitle,
+		SourceDesc:        sourceDesc,
+		GatewayResourceID: resourceName + ".gateway_route_id",
+	}, resourceOpenTelemetryMetricsSourceConfigTpl)
+	if err != nil {
+		t.Fatalf("error parsing config template: %s", err)
+	}
+
+	configGatewayUpdateErr, err := ParsedAccConfig(resourceOpenTelemetryMetricsSourceConfig{
+		PipelineName:      pipelineName,
+		PipelineTitle:     pipelineTitle,
+		SourceName:        sourceName,
+		SourceTitle:       sourceTitle,
+		SourceDesc:        sourceDesc,
+		GatewayResourceID: resourceNamePipeline + ".id",
+	}, resourceOpenTelemetryMetricsSourceConfigTpl)
+	if err != nil {
+		t.Fatalf("error parsing config template: %s", err)
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		PreCheck:                 func() { TestPreCheck(t) },
+		Steps: []resource.TestStep{
+			// Error: pipeline_id is required
+			{
+				Config: `
+					resource "mezmo_pipeline" "test_parent" {
+						title = "parent pipeline"
+					}
+					resource "mezmo_open_telemetry_traces_source" "my_source" {
+						title = "my kafka title"
+						description = "my kafka description"
+					}`,
+				ExpectError: regexp.MustCompile("The argument \"pipeline_id\" is required"),
+			},
+			// Create and Read testing
+			{
+				Config: GetProviderConfig() + config,
+				Check:  checkResourceOpenTelemetryMetricsSource(resourceName),
```

Comment:
Same comment--the use of a function here feels unnecessary and adds indirection.